### DISCUSSION TITLE
Add `"""` quote-style, like Python does

### DISCUSSION THREAD

**Discussion by Tishj** [duckdb/duckdb#6780] [Ideas]
This is another thing I have really come to appreciate from python, and something that would also be incredibly helpful in DuckDB when dealing with strings that contains quotes

For those unfamiliar with the functionality:
```py
"""
this can contain "any" number of "quoted" strings, also 'single' quoted 'strings', without terminating
"""
```

Since we likely don't want to support escaping, as that is not a thing in SQL, I think this would be a really good alternative

**Comment by Alex-Monahan** [comment]
JavaScript uses back ticks for multi-line strings if that is easier to implement!

**Comment by tbeason** [comment]
Julia does the same.

**Comment by douenergy** [comment]
+1

**Comment by Tishj** [comment]
```
D select ```
  this can contain "any" number of "quoted" strings, also 'single' quoted 'strings', without terminating
  ```;
┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ '\nthis can contain "any" number of "quoted" strings, also ''single'' quoted ''strings'', without terminating\n' │
│                                                     varchar                                                      │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ \nthis can contain "any" number of "quoted" strings, also 'single' quoted 'strings', without terminating\n       │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
D 
```

Any interest in this still?
It was actually easier than expected 😅

**Reply by Tishj** [reply]
Some other tests, unterminated and containing multi lines:
```
D select ```
  test
  test;
Parser Error:
Unterminated triple backtick string literal at or near "```
test
test;"

LINE 2: ```
        ^
D select ```
  test
  test```;
┌──────────────────┐
│ 'test\ntest' │
│     varchar      │
├──────────────────┤
│ test\ntest   │
└──────────────────┘
```

**Reply by Tishj** [reply]
https://github.com/Tishj/duckdb/tree/triple_backtick_literal

**Reply by LennartH** [reply]
Are the triple backticks really necessary? I thought any string literal could be multiline strings already. But backticks as an alternative string literal syntax would be handy.

**Comment by vlulla** [comment]
I am curious, what is the advantage of using triple quoted strings over dollar quoted strings?

```sql
select $mystring$
this can contain "any" number of "quoted" strings, also 'single' quoted 'strings', without terminating
$mystring$; 
```

This will work in duckdb (just like in PostgreSQL)! What am I missing?
